src/renderer/vsync/macos_display_link.rs

```diff
@@ -207,19 +204,22 @@ impl<UserData> Drop for MacosDisplayLink<UserData> {
 
 // Here is the doc about how to do this. https://developer.apple.com/documentation/appkit/nsscreen/1388360-devicedescription?language=objc
 pub fn get_display_id_of_window(window: &Window) -> core_video::CGDirectDisplayID {
-    let mut result = 0;
-    autoreleasepool(|| unsafe {
-        let key: id = NSString::alloc(nil)
-            .init_str("NSScreenNumber")
-            .autorelease();
-        if let RawWindowHandle::AppKit(handle) = window.raw_window_handle() {
-            let ns_window: id = handle.ns_window as id;
-            let display_id_ns_number = ns_window.screen().deviceDescription().valueForKey_(key);
-            result = msg_send![display_id_ns_number, unsignedIntValue];
-        } else {
-            // Should be impossible.
-            panic!("Not an AppKitWindowHandle.")
-        }
-    });
-    result
+    unsafe fn get_display_id(window: &Window) -> Option<core_video::CGDirectDisplayID> {
+        autoreleasepool(|_pool| {
+            let key: Id<NSString> = NSString::from_str("NSScreenNumber");
+            if let RawWindowHandle::AppKit(handle) = window.raw_window_handle() {
+                let ns_window = Id::retain(handle.ns_window as *mut NSWindow)?;
+                let screen = ns_window.screen()?;
+                let descr = screen.deviceDescription();
+                let display_id_ns_number = descr.get(&key)?;
+                let res = msg_send![display_id_ns_number, unsignedIntValue];
+                let _ = Id::autorelease(ns_window, _pool);
```

Comment:
I believe it's a mistake to autorelease ns_window here. Actually I think we should remove the autoreleasepool entirely. Because icrate should have already dealt with the ref count correctly.